pkg/localstore/reserve.go

```diff
@@ -164,3 +164,37 @@ func withinRadius(db *DB, item shed.Item) bool {
 	po := db.po(swarm.NewAddress(item.Address))
 	return po >= item.Radius
 }
+
+func (db *DB) ComputeReserveSize(startPO uint8) (uint64, error) {
+
+	var count uint64
+
+	err := db.pullIndex.Iterate(func(item shed.Item) (stop bool, err error) {
+		count++
+		return false, nil
+	}, &shed.IterateOptions{
+		StartFrom: &shed.Item{
```

Comment:
Lets add some comments here on why this gives the whole reserve in this iteration.

Reply:
see the function comment

Reply:
For nodes with lots of locally pinned data, iterating an index just to count things is very expensive.  I know this from the experience of watching how long it takes the pull cursors to be calculated on my 1.3TB node.  This node is currently taking just under an HOUR to handle a cursors request.  I have added a throttle to only allow a single request to be handled at a time to protect the CPU and disk from extreme overload.  I don't recall what index the cursor determination uses, but it's in the chain from pullsync to cursorHandler to the Cursors() invocation.

Reply:
I added an elapsed time log around ComputeReserveSize on my 1.3TB node and here's the first two passes:
```"time"="2022-09-23 14:46:48.126496" "level"="debug" "logger"="node/localstore" "v"=1 "msg"="ComputeReserveSize executed" "po"=12 "elapsed"="3m28.282063599s" "count"=82552
"time"="2022-09-23 14:46:48.127805" "level"="debug" "logger"="node/depthmonitor" "msg"="depthmonitor: state" "current size"=82552 "radius"=12 "chunks/sec rate"=8.024555244444445
"time"="2022-09-23 14:54:02.156451" "level"="debug" "logger"="node/localstore" "v"=1 "msg"="ComputeReserveSize executed" "po"=12 "elapsed"="2m14.028099138s" "count"=82555
"time"="2022-09-23 14:54:02.156488" "level"="debug" "logger"="node/depthmonitor" "msg"="depthmonitor: state" "current size"=82555 "radius"=12 "chunks/sec rate"=4.99```